---
title: "Project proposal"
author: "Judy Malas"
output: html_document
date: November 17th 2021
---

```{r load-packages, message = FALSE}
library(tidyverse)
library(broom)
library(phyloseq)
```

#### 1. Data

The dataset that will be used comes from an experiment conducted to test the effects of chemical additives on the microbial populations within landfills. 

The study used simulated landfill microcosms and tracked microbial populations over a 65 day period. Fifteen microcosms were set up identically on day 0, and DNA was sampled from all 15 microcosms on days 6 (T1), 20 (T2), 29 (T3), and 40 (T4). On day 40, three microcosms were destructively sampled for chemical measurements including sulfide, sulfate, total iron, and ferrous iron. 
On day 44, triplicate microcosms were "spiked" with different additives (sodium sulfate, antibiotics, and Fe(OH)3). Two additional DNA samples were taken at 50 (T5) days and 65 (T6) days of incubation from the remaining 12 microcosms. At the end of the 65 day period, all microcosms were destructively sampled for chemical concentrations of sulfide, sulfate, total iron, and ferrous iron. 

After sampling, DNA was extracted and a portion of the 16S rRNA gene was amplified and sequenced. The initial phase of data processing for 16s data involves quality control, demultiplexing, and binning sequences into amplicon sequence variants (ASVs -- also known as OTUs or operational taxonomic units). 
The ASVs are then matched to a reference library to identify their taxonomy. 

This markdown picks up after this initial phase of data processing. In the following chunk, we have a "Phyloseq object" which was made in the phyloseq package in R. The object consists of an OTU table (ASVs abundance matched to samples); a metadata file ("sample_data") which includes the time the sample was taken, spike added to the sample, etc.; a taxa_table which identifies which ASV is matched to which taxonomy by 6 taxonomic ranks (i.e. Kingdom, Phylum, Class, Order, Family, Genus); and an optional phylogentic tree.

```{r read_phyloseq_object}
full_16s_dataset = readRDS("~/Desktop/16s_script/all")
full_16s_dataset
```

In order to analyze this data with the tidyverse functions, we need to convert the phylsoeq object to a regular data frame. First we use the "transform_sample_counts" function to change our absolute counts into relative count, which are more useful for visualization than absolute counts. We can then use the function "psmelt" which will combine all the elements of the object, except the phylogentic tree, into a single data frame. 
```{r create-dataframe-with-psmelt}
full_16s_dataset_relative = transform_sample_counts(full_16s_dataset, function(x) {x/sum(x)})
full_16s_df = psmelt(full_16s_dataset_relative)  
glimpse(full_16s_df)

```

The data contain `r nrow(full_16s_df)` rows and `r ncol(full_16s_df)` columns. The data are relatively tidy because each row representations one observation of an OTU's relative abundance in each DNA sample. 
However, the column names are not very tidy; some are capitalized and some are not and there are random punctuation marks in some column names. Additionally, one column is completely blank and should be deleted. 

#### 2. Exploratory Visualization

```{r histogram-rel-abundance}
ggplot(full_16s_df, aes(Abundance))+
  geom_histogram()+
  labs(x= "Relative Abundance")
```
From the plot above we can see that most of the samples have OTUs with very low relative abundances <002. Perhaps from this plot we can get an understanding of how much of the data gets cut off when we filter for different abundances.


Differences at the class level at the end of the experiment? 
```{r trial-vis}

full_16s_df %>% 
  filter(sample_time == "T6", Phylum == "Firmicutes", Abundance > 0.0001) %>% 
    ggplot(aes(x = spike, y = Abundance, fill = Class))+
    geom_bar(stat = "identity")+ 
  labs(y = "Relative Abundance")
  
```
Overall, the relative abundance of the different classes remain fairly similar across the experiments. 
However, the Fe(OH)3 spike seems to have some additional classes not seen in the other experiments (Moorellia, Negativicutes, and Thermacetogenia), at least at this filtering level. The y axis here goes until 3 because there are three bottles in each spike.


```{r group-by}
genus_desending = full_16s_df %>% 
  group_by(Genus) %>% 
  count() %>% 
  arrange(desc(n))

genus_desending

  ggplot(genus_desending, aes(n)) +
  geom_histogram()


```
There are 127 identified Genuses, are most of them are observed <10000 times, with one outlier (Caproiciproducens). That being said, the vast majority of Genuses are NA. 

#### 3. Possible question and method
The obvious question, and the reason the experiment was conducted is: 
How do the different additives tested effect the microbial community in a landfill? 

Using the tools we learned in class, it would make sense to dig into each of the different taxonomic groups at some level and search for visual differences from the plots. For example, I could use group_by and summarize to make averages for triplicates across different taxonomic levels by filtering for different Taxa.
  
